From 935a16ee910d880d91fdaad269e95ea65359e879 Mon Sep 17 00:00:00 2001
From: Terrence Pu <terrence.pu@amlogic.com>
Date: Mon, 27 Jul 2020 12:22:06 -0700
Subject: [PATCH 5/5] RDK: Force to use PQ_BOX to supprot STR_2_STR mode

PD#SWPL-26565

Problem:
After STR_2_STR resume the TV show green display because wb_enable

Solution:
disabe wb from PQ_Controller and SysFs access

Verify:
u212

Change-Id: Iace678d7fdf1f154da07923e938206c2faad1b95
Signed-off-by: Terrence Pu <terrence.pu@amlogic.com>
---
 drivers/amlogic/media/enhancement/amvecm/amve.c   | 8 ++++++++
 drivers/amlogic/media/enhancement/amvecm/amvecm.c | 9 ++++++---
 2 files changed, 14 insertions(+), 3 deletions(-)

diff --git a/drivers/amlogic/media/enhancement/amvecm/amve.c b/drivers/amlogic/media/enhancement/amvecm/amve.c
index c7040c4..23aaa69 100644
--- a/drivers/amlogic/media/enhancement/amvecm/amve.c
+++ b/drivers/amlogic/media/enhancement/amvecm/amve.c
@@ -2120,6 +2120,7 @@ void set_gamma_regs(int en, int sel)
 
 void amvecm_wb_enable(int enable)
 {
+#if 0
 	if (enable) {
 		wb_en = 1;
 		if (video_rgb_ogo_xvy_mtx)
@@ -2133,6 +2134,13 @@ void amvecm_wb_enable(int enable)
 		else
 			WRITE_VPP_REG_BITS(VPP_GAINOFF_CTRL0, 0, 31, 1);
 	}
+#else // disable wb_enable for OTT product on sysmode support
+	wb_en = 0;
+	if (video_rgb_ogo_xvy_mtx)
+		WRITE_VPP_REG_BITS(VPP_MATRIX_CTRL, 0, 6, 1);
+	else
+		WRITE_VPP_REG_BITS(VPP_GAINOFF_CTRL0, 0, 31, 1);
+#endif
 }
 
 int vpp_pq_ctrl_config(struct pq_ctrl_s pq_cfg)
diff --git a/drivers/amlogic/media/enhancement/amvecm/amvecm.c b/drivers/amlogic/media/enhancement/amvecm/amvecm.c
index c6a127a..55d011e 100644
--- a/drivers/amlogic/media/enhancement/amvecm/amvecm.c
+++ b/drivers/amlogic/media/enhancement/amvecm/amvecm.c
@@ -7731,15 +7731,18 @@ void init_pq_control(unsigned int enable)
 /* #if (MESON_CPU_TYPE == MESON_CPU_TYPE_MESONG9TV) */
 void init_pq_setting(void)
 {
+#if 0
 	struct vinfo_s *vinfo = get_current_vinfo();
-
+#endif
 	int bitdepth;
-
+#if 0
 	if (vinfo->viu_color_fmt == COLOR_FMT_RGB444)
 		init_pq_control(PQ_TV);
 	else
 		init_pq_control(PQ_BOX);
-
+#else //only enable PQ_BOX for OTT product. Why PQ_TV setting is used  when hdmimode = null?
+	init_pq_control(PQ_BOX);
+#endif
 	/*ai pq interface*/
 	ai_detect_scene_init();
 	adaptive_param_init();
-- 
2.7.4

